library/core/src/fmt/mod.rs

```diff
@@ -315,7 +315,7 @@ macro_rules! arg_new {
         #[doc(hidden)]
         #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
         #[inline]
-        pub fn $f<'b, T: $t>(x: &'b T) -> ArgumentV1<'_> {
+        pub const fn $f<'b, T: $t>(x: &'b T) -> ArgumentV1<'_> {
```

Comment:
Unless there's an existing pull request I haven't spotted, I'm expecting to need to spin this and src/test/ui/consts/const-eval/format.* off to a second pull request.

Reply:
This doesn't actually work (the const panic formatting code breaks) so I had to do the formatting by hand using `fmt_u64` from `fmt/num.rs`.